src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,209 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a list of tasks.
+ */
+public class TaskList {
+    // The list of tasks
+    private List<Task> tasks;
+    // The formatter for the date and time
+    private final DateTimeFormatter inputFormatter;
+
+    /**
+     * A constructor for the task list if an initial list exist
+     *
+     * @param tasks the list of tasks
+     */
+    public TaskList(List<Task> tasks) {
+        this.tasks = tasks;
+        this.inputFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+    }
+
+    /**
+     * A constructor for the task list if an initial list does not exist
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+        this.inputFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+    }
+
+    /**
+     * Returns the number of tasks in the list.
+     *
+     * @return the number of tasks
+     */
+    public int getNumberOfTasks() {
+        return this.tasks.size();
+    }
+
+    /**
+     * Returns the list of tasks.
+     *
+     * @return the list of tasks
+     */
+    public List<Task> getTasks() {
+        return this.tasks;
+    }
+
+    /**
+     * Returns a success message when a task has been successfully added.
+     *
+     * @return the success message
+     */
+    public String printAddTaskSuccessMessage() {
+        StringBuilder message = new StringBuilder();
+        message.append("Got it. I've added this task:\n");
+        message.append(" ");
+        message.append(this.tasks.get(this.tasks.size() - 1).toString());
+        message.append("\n");
+        message.append("Now you have ");
+        message.append(this.tasks.size());
+        message.append(" task(s) in the list.");
+        return message.toString();
+    }
+
+    /**
+     * Returns a success message when a task has been successfully deleted.
+     *
+     * @return the success message
+     */
+    public String printRemoveTaskSuccessMessage(Task task) {
+        StringBuilder message = new StringBuilder();
+        message.append("Noted. I've removed this task:\n");
+        message.append(" ");
+        message.append(task.toString());
+        message.append("\n");
+        message.append("Now you have ");
+        message.append(this.tasks.size());
+        message.append(" task(s) in the list.");
+        return message.toString();
+    }
+
+    /**
+     * Takes in the task details and add the task to the list.
+     *
+     * @param task the task description
+     * @param typeOfTask the type of the task to be added
```

Comment:
One could consider renaming `typeOfTask` to just `taskType`, which is simpler without losing its descriptive quality